tools/mdfileconverter/md_converter.xml

```diff
@@ -0,0 +1,124 @@
+<tool id="md_converter" name="MDTraj file converter" version="1.9.1">
+    <macros>
+        <xml name='traj' token_format='format'>
+            <when value="@FORMAT@">
+                <param argument="input_file" type="data" format='@FORMAT@' label="@FORMAT@ input file for conversion"/>
+                <param argument="output_format" type="select" label="Output">
+                    <option value="trr">TRR file</option>
+                    <option value="xtc">XTC file</option>
+                    <option value="dcd">DCD file</option>
+                </param>
+            </when>
+        </xml>
+        <xml name='str' token_format='format'>
+            <when value="@FORMAT@">
+                <param argument="input_file" type="data" format='@FORMAT@' label="@FORMAT@ input file for conversion"/>
+                <param argument="output_format" type="select" label="Output">
+                    <option value="gro">GRO file</option>
+                    <option value="pdb">PDB file</option>
+                </param>
+            </when>
+        </xml>
+    </macros>
+
+    <description>- interconvert between MD trajectory or structure file formats.</description>
+
+    <requirements>
+        <requirement type="package" version="1.9.1">mdtraj</requirement>
+        <requirement type="package" version="2018.2">gromacs</requirement>
+    </requirements>
+
+    <command detect_errors="exit_code"><![CDATA[
+
+        #if $input.input_format == 'gro' or $input.input_format == 'pdb':
+            ln -s '$input_file' ./input.$input.input_format &&
+            gmx editconf -f ./input.$input.input_format -o ./output.${input.output_format} &&
+            cp ./output.${input.output_format} $output
+
+        #else
+            ln -s '$input_file' ./input.$input.input_format &&
+            mdconvert ./input.$input.input_format -o ./output.${input.output_format} &&
+            cp ./output.${input.output_format} $output
+        
+        #end if
+
+
+    ]]></command>
+    <inputs>
+        <conditional name="input">
+            <param name="input_format" type="select" label="File format of input">
```

Comment:
If you do not add additional parameters to the single files you could remove the conditional I think, see above.

Reply:
Ah I see, not everything is convertible ....